foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/actions/FilterAction.java

```diff
@@ -145,13 +151,108 @@ public SearchRequest getRequestBuilder(Query parameter) {
+                .searchType(SearchType.QUERY_THEN_FETCH)
+                .source(new SearchSourceBuilder()
+                                .timeout(new TimeValue(getGetQueryTimeout(), TimeUnit.MILLISECONDS))
+                                .size(parameter.getLimit())
+                                .query(new ElasticSearchQueryGenerator().genFilter(parameter.getFilters()))
+                                .sort(Utils.storedFieldName(parameter.getSort().getField()),
+                                      ResultSort.Order.desc == parameter.getSort().getOrder()
+                                      ? SortOrder.DESC : SortOrder.ASC));
+        searchRequest.scroll(TimeValue.timeValueSeconds(elasticsearchTuningConfig.getScrollTimeInSeconds()));
+        return searchRequest;
+    }
+
+    private ActionResponse executeScrollRequest(Query parameter) {
+        List<String> ids = new ArrayList<>();
+        String scrollId;
+        long totalHits = 0;
+        try {
+            if (StringUtils.isNotEmpty(parameter.getScrollId())) {
+                scrollId = parameter.getScrollId();
+                SearchScrollRequest scrollRequest = new SearchScrollRequest(scrollId);
+                scrollRequest.scroll(TimeValue.timeValueSeconds(elasticsearchTuningConfig.getScrollTimeInSeconds()));
+                SearchResponse searchScrollResponse = getConnection().getClient().scroll(scrollRequest,
+                                                                                         RequestOptions.DEFAULT);
+                scrollId = searchScrollResponse.getScrollId();
+                SearchHits hits = searchScrollResponse.getHits();
+                for (SearchHit searchHit : hits) {
+                    ids.add(searchHit.getId());
+                }
+            }
+            else {
+                SearchRequest searchRequest = getScrollRequestBuilder(parameter);
+                SearchResponse searchResponse = getConnection()
+                        .getClient()
+                        .search(searchRequest, RequestOptions.DEFAULT);
+                SearchHits searchHits = searchResponse.getHits();
+                totalHits = searchHits.getTotalHits();
+                for (SearchHit searchHit : searchHits) {
+                    ids.add(searchHit.getId());
+                }
+                scrollId = searchResponse.getScrollId();
+            }
+            return getResponse(parameter, totalHits, ids, scrollId);
+        }
+        catch (IOException e) {
+            throw FoxtrotExceptions.createQueryExecutionException(parameter, e);
+        }
+    }
+
+    private QueryResponse getResponse(Query parameter, long totalHits, List<String> ids, String scrollId) {
+        if (ids.isEmpty()) {
+            return QueryResponse
+                    .builder()
+                    .documents(Collections.emptyList())
+                    .totalHits(0)
+                    .scrollResponse(true)
```

Comment:
what makes sense is to remove this variable and create one called moreDataAvailable. Set it to false in this case, true in the other.